### PR TITLE
sipify layer tree classes

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: python/core/layertree/qgslayertreenode.sip [hunk 1]
```diff
@@ -1,194 +1,314 @@
...
+%End
+    void nameChanged( QgsLayerTreeNode *node, QString name );
+%Docstring
+.. versionadded:: 3.0
+%End
 
   protected:
 
-    QgsLayerTreeNode( NodeType t );
+    QgsLayerTreeNode( NodeType t, bool checked = true );
```
Comment (3nids) [inline python/core/layertree/qgslayertreenode.sip]: is the extra argument ok, or was it meant not to be used in Python?

File: python/core/layertree/qgslayertreenode.sip [hunk 2]
```diff
@@ -1,194 +1,314 @@
...
+%Docstring
+Low-level removal of children from the node.
+%End
 
-  private:
-    QgsLayerTreeNode( const QgsLayerTreeNode& other );
+  protected:
 
+  private:
+    QgsLayerTreeNode( const QgsLayerTreeNode &other );
```
Comment (3nids) [inline python/core/layertree/qgslayertreenode.sip]: is this expected?

File: python/core/layertree/qgslayertree.sip [hunk 3]
```diff
@@ -1,61 +1,193 @@
...
-    //! Cast node to a group. No type checking is done - use isGroup() to find out whether this operation is legal.
-    // PYTHON: automatic cast
-    //inline QgsLayerTreeGroup* toGroup( QgsLayerTreeNode* node );
 
-    //! Cast node to a layer. No type checking is done - use isLayer() to find out whether this operation is legal.
-    // PYTHON: automatic cast
-    //inline QgsLayerTreeLayer* toLayer( QgsLayerTreeNode* node );
+
+
+    QgsLayerTree();
```
Comment (3nids) [inline python/core/layertree/qgslayertree.sip]: this one and next one were not present in old sip file. there is no comment (e.g not available in python bindings). Should they be present?

File: python/core/layertree/qgslayertreenode.sip (hunk 1 above)
Comment (wonder-sk) [inline python/core/layertree/qgslayertreenode.sip]: Yeah the extra argument should be fine - probably omitted earlier

File: python/core/layertree/qgslayertreenode.sip (hunk 2 above)
Comment (wonder-sk) [inline python/core/layertree/qgslayertreenode.sip]: Not sure if I understand this change - earlier in the header file the copy constructor of QgsLayerTreeNode is declared as protected ... why would we want to declare it in SIP as private?

File: python/core/layertree/qgslayertree.sip (hunk 3 above)
Comment (wonder-sk) [inline python/core/layertree/qgslayertree.sip]: The default constructor should be in SIP as well - I think SIP creates a default constructor if no other constructor is specified.

The copy constructor actually should be probably protected or private in the header - we have clone() method instead...

File: python/core/layertree/qgslayertreenode.sip (hunk 2 above)
Comment (3nids) [inline python/core/layertree/qgslayertreenode.sip]: not sure, it was defined like that in the older SIP file, so I kept the same (I didn't noticed it was also declared as protected). So removing this.

File: python/core/layertree/qgslayertree.sip (hunk 3 above)
Comment (3nids) [inline python/core/layertree/qgslayertree.sip]: I made the copy constructor private

File: python/core/layertree/qgslayertreenode.sip (hunk 1 above)
Comment (3nids) [inline python/core/layertree/qgslayertreenode.sip]: ok, thanks

**Comment by sbrunner** [comment]
Is it a good idea to commit auto-generated files?

**Comment by 3nids** [comment]
@sbrunner while the script to generate them is stll not very stable, yes. Once we have moved all the files to automatic generation, it would indeed make sense to generate them on building.

**Comment by sbrunner** [comment]
OK, thanks for your response :-)

**Pull request merged**